framework/src/main/java/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -1524,6 +1521,112 @@ protected void performFlowAnalysisForClass(ClassTree classTree) {
+   * @param fieldValues values of fields to be used
+   * @param capturedStore the input Store to use for captured variables, e.g. in a lambda
+   */
+  private void performFlowAnalysisForMethod(
+      ClassTree classTree,
+      CFGMethod method,
+      Queue<IPair<ClassTree, @Nullable Store>> classQueue,
+      List<FieldInitialValue<Value>> fieldValues,
+      @Nullable Store capturedStore) {
+    // The list contains one element for each `return` statement in the lambda (the map key).
+    Map<LambdaExpressionTree, List<AnnotationMirrorSet>> lambdaResultTypeMap = new HashMap<>();
+    Map<LambdaExpressionTree, ControlFlowGraph> lambdaToCFG = new HashMap<>();
+    ControlFlowGraph methodCFG = null;
+
+    boolean isConstructor = TreeUtils.isConstructor(method.getMethod());
+
+    // Analyze `method` and all lambdas contained in `method` until the type of the lambda result
+    // expressions do not change.
+    while (true) {
+      Queue<IPair<ClassTree, @Nullable Store>> classQueueInMethod = new ArrayDeque<>();
+      Queue<IPair<LambdaExpressionTree, @Nullable Store>> lambdaQueueInMethod = new ArrayDeque<>();
+      methodCFG =
+          analyze(
+              classQueueInMethod,
+              lambdaQueueInMethod,
+              method,
+              fieldValues,
+              methodCFG,
+              isConstructor,
+              /* updateInitializationStore= */ false,
+              /* isStatic= */ false,
+              capturedStore);
+
+      boolean anyLambdaResultChanged = false;
+      while (!lambdaQueueInMethod.isEmpty()) {
+        IPair<LambdaExpressionTree, @Nullable Store> lambdaPair = lambdaQueueInMethod.remove();
+        LambdaExpressionTree lambda = lambdaPair.first;
+        MethodTree mt =
+            (MethodTree) TreePathUtil.enclosingOfKind(getPath(lambda), Tree.Kind.METHOD);
+        ControlFlowGraph cfgLambda =
+            analyze(
+                classQueueInMethod,
+                lambdaQueueInMethod,
+                new CFGLambda(lambda, classTree, mt),
+                fieldValues,
+                lambdaToCFG.get(lambda),
+                /* isInitializationCode= */ false,
+                /* updateInitializationStore= */ false,
+                /* isStatic= */ false,
+                lambdaPair.second);
+        lambdaToCFG.put(lambda, cfgLambda);
+
+        List<AnnotationMirrorSet> returnedExpressionAnnos =
+            CollectionsPlume.mapList(
+                tree -> getAnnotatedType(tree).getPrimaryAnnotations(),
+                TreeUtils.getReturnedExpressions(lambda));
+        List<AnnotationMirrorSet> prevReturnedExpressionAnnos = lambdaResultTypeMap.get(lambda);
+        if (prevReturnedExpressionAnnos == null
+            || !prevReturnedExpressionAnnos.equals(returnedExpressionAnnos)) {
+          anyLambdaResultChanged = true;
+        }
+        lambdaResultTypeMap.put(lambda, returnedExpressionAnnos);
+      }
+
```

Comment:
I think that after this point, the keyset of `lambdaToCFG` (and `lambdaToResultTypes`) never changes.  Is that true?  If true, does `containsAllVoidLambas` on the next line need to be repeatedly computed?